test/support/test_helpers.ex

```diff
@@ -1,39 +1,65 @@
 defmodule RemoteRetro.TestHelpers do
   use Wallaby.DSL
-  alias RemoteRetro.{Repo, User, Vote}
+  alias RemoteRetro.{Repo, User, Vote, Idea, Participation}
 
   @mock_user Application.get_env(:remote_retro, :mock_user)
 
-  def persist_idea_for_retro(context) do
-    %{idea: idea, retro: retro, user: user} = context
-
-    idea = if idea.category == "action-item" do
-            Map.merge(idea, %{retro_id: retro.id, user_id: user.id, assignee_id: user.id})
-            |>Repo.insert!
-          else
-            Map.merge(idea, %{retro_id: retro.id, user_id: user.id})
-            |>Repo.insert!
-          end
-
-    Map.put(context, :idea, idea)
-  end
-
-  def use_all_votes(%{user: user, idea: idea} = context) do
+  def use_all_votes(%{test_user: user, idea: idea} = context) do
     now = DateTime.utc_now
     vote = [user_id: user.id, idea_id: idea.id, inserted_at: now, updated_at: now]
     Repo.insert_all(Vote, [vote, vote, vote, vote, vote])
     context
   end
 
-  def persist_user_for_retro(context) do
-    context = Map.merge(%{user: @mock_user}, context)
-    %{user: user} = context
+  defp user_name_atom(name) do
+    String.replace(name, ~r/ +/, "") |> Macro.underscore |> String.to_atom
+  end
+
+  defp user_map(users) do
+    Enum.reduce users, %{}, fn user, acc ->
+      Map.put(acc, user_name_atom(user.name), user)
+    end
+  end
+
+  defp persist_user(user) do
     user_params = User.build_user_from_oauth(user)
-    user =
-      User.changeset(%User{}, user_params)
-      |> Repo.insert!
+    User.changeset(%User{}, user_params)
+        |> Repo.insert
+  end
+
+  def persist_users_for_retro(%{users: users} = context) do
+    Enum.each(users, fn user ->
+      persist_user(user)
+    end)
+    persisted_users = User |> Repo.all 
+    Map.merge(user_map(persisted_users), context)
+  end
+
+  defp persist_assigned_idea(user, idea, retro) do
+    %Participation{retro_id: retro.id, user_id: user.id} |> Repo.insert!
```

Comment:
Hmmm... our application adds a `participation` record when the user joins a retro, so I think that the participation record should be persisted in the `persist_users_for_retro` helper, as it will confused anyone who doesn't have intimate knowledge of the system would be confused as to why a participation is being created as an implicit pre-req to the action item being assigned

Reply:
💯